scripts/templates/generate_coverage_report

```diff
@@ -125,24 +129,38 @@ if [[ "$1" == "--codecov" ]]; then
       echo "WARNING: No CODECOV_TOKEN defined. Upload to CodeCov might fail"
    fi
 
-   # Remove other files to be sure CodeCov uses only the coverage.info file
-   rm -f lcov.info
-   remove_gcda_files
-
    # Upload report to CodeCov
+   if [ ! -d build/tmp ]; then
+      mkdir build
+      mkdir build/tmp
+   fi
 
-   if ! bash <(curl -s https://codecov.io/bash) -X gcov; then
+   cd build/tmp/
 
-      echo "ERROR: codecov did not collect coverage reports"
-      echo "Trying again in 1 minute..."
-      sleep 60
+   if [ ! -f codecov ]; then
+
+      curl -Os https://uploader.codecov.io/latest/linux/codecov
+      if [ ! -f codecov ]; then
+         echo "ERROR: codecov did not collect coverage reports"
+         echo "Trying again in 1 minute..."
+         sleep 60
+      fi
 
-      if ! bash <(curl -s https://codecov.io/bash) -X gcov; then
+      curl -Os https://uploader.codecov.io/latest/linux/codecov
+      if [ ! -f codecov ]; then
          echo "FATAL ERROR: Codecov cannot collect the coverage reports"
          exit 1
       fi
+
    fi
 
+   chmod +x codecov
+
+   if [ ! -z  $"CODECOV-TOKEN" ]; then
+      ./codecov
+   else
+      ./codecov -t $"CODECOV-TOKEN"
```

Comment:
**BUG:** First, you don't need to do that at all. If the env variable has been exported, the codecov tool will see it.
Also, the name was `CODECOV_TOKEN`. Third, $"something" doesn't mean anything to me. "$something" is the correct form.

Reply:
**BUG:** the -X gcov option is missing.

Reply:
So i just need to do `$BUILD/tmp/codecov -X gcov`?

Reply:
Yes, of course. The whole point was to replace `bash <(curl -s https://codecov.io/bash)` with `$BUILD/tmp/codecov` everywhere after downloading it.

Reply:
Perfect!